src/ctia/schemas/core.clj

```diff
@@ -38,6 +43,144 @@
    :created java.util.Date
    (s/optional-key :modified) java.util.Date})
 
+(s/defschema RealizeFnServices
+  "Maps of service functions available for realize-fns"
+  {:ConfigService {:get-in-config (s/=>* s/Any
+                                         [[s/Any]]
+                                         [[s/Any] s/Any])}
+   :StoreService {:read-store (s/pred ifn?)} ;;varags
+   :GraphQLNamedTypeRegistryService
+   {:get-or-update-named-type-registry
+    (s/=> graphql.schema.GraphQLType
+          s/Str
+          (s/=> graphql.schema.GraphQLType))}
+   :IEncryption {:encrypt (s/=> s/Any s/Any)
+                 :decrypt (s/=> s/Any s/Any)}})
+
+(s/defn APIHandlerServices->RealizeFnServices
+  :- RealizeFnServices
+  [services :- APIHandlerServices]
+  (select-keys-in
+    services
+    [:ConfigService [:get-in-config]]
+    [:StoreService [:read-store]]
+    [:GraphQLNamedTypeRegistryService [:get-or-update-named-type-registry]]
+    [:IEncryption [:decrypt :encrypt]]))
+
+(s/defschema GraphQLRuntimeOptions
+  "A map of options to resolve a DelayedGraphQLValue"
+  {:services RealizeFnServices})
```

Comment:
I made this an extensible map argument instead of just a services arg since we can only do this "delay by returning function" trick once, and if we need additional non-service values it'll come in really handy.

Reply:
So `GraphQLRuntimeOptions` contains all the services graph.
And It could probably contain other things added during the init.
So I think it looks pretty similar of the idea of ServiceContext to me because it looks used the same way.

So maybe `GraphQLRuntimeContext` because with `Options` it doesn't sound like the Options contains handlers.

Reply:
Ah yes!